src/heretic/main.py

```diff
@@ -498,12 +580,28 @@ def objective_wrapper(trial: Trial) -> tuple[float, float]:
                         private = visibility == "Private"
 
                         print("Uploading model...")
+                        strategy = obtain_merge_strategy(settings)
+                        if strategy is None:
+                            print("[yellow]Action cancelled.[/]")
+                            continue
+
+                        if strategy == "adapter":
+                            model.model.push_to_hub(
+                                repo_id,
+                                private=private,
+                                token=token,
+                            )
+                        else:
+                            merged_model = model.get_merged_model()
+                            merged_model.push_to_hub(
+                                repo_id,
+                                private=private,
+                                token=token,
+                            )
+                            del merged_model
+                            gc.collect()
+                            empty_cache()
```

Comment:
One of these lines is redundant, because `empty_cache()` implies `gc.collect()`.